src/package1/package1Version.ts

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as os from 'os';
+import { Connection, Lifecycle, Messages, PollingClient, StatusResult } from '@salesforce/core';
+import { Duration } from '@salesforce/kit';
+import {
+  IPackageVersion1GP,
+  Package1Display,
+  Package1VersionCreateRequest,
+  Package1VersionEvents,
+  PackagingSObjects,
+} from '../interfaces';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/packaging', 'package1Version');
+
+/**
+ * Package1Version class - Class to be used with 1st generation package versions
+ *
+ * implementation examples can be seen here: https://github.com/salesforcecli/plugin-packaging/tree/main/src/commands/force/package1/
+ */
+export class Package1Version implements IPackageVersion1GP {
+  public constructor(private connection: Connection) {}
+
+  /**
+   * Returns the status of a PackageUploadRequest
+   *
+   * @param connection Connection to the target org
+   * @param id 0HD Id of the PackageUploadRequest
+   */
+  public static async getCreateStatus(
+    connection: Connection,
+    id: string
+  ): Promise<PackagingSObjects.PackageUploadRequest> {
+    if (!id.startsWith('0HD')) {
+      throw messages.createError('invalid0HDId', [id]);
+    }
+    return (await connection.tooling
+      .sobject('PackageUploadRequest')
+      .retrieve(id)) as unknown as PackagingSObjects.PackageUploadRequest;
+  }
+
+  /**
+   * Executes server-side logic for the package1:display command
+   *
+   * @param connection Connection to the org
+   * @param id: id of the MetadataPackageVersion sObject (starts with 04t)
+   */
+  public static async display(connection: Connection, id: string): Promise<Package1Display[]> {
+    if (!id.startsWith('04t')) {
+      throw messages.createError('invalid04tId', [id]);
+    }
+    const query = `SELECT Id,MetadataPackageId,Name,ReleaseState,MajorVersion,MinorVersion,PatchVersion,BuildNumber FROM MetadataPackageVersion WHERE id = '${id}'`;
+    const results = (await connection.tooling.query<PackagingSObjects.MetadataPackageVersion>(query)).records;
+    return results.map((result) => ({
```

Comment:
Why not return the entire object. This function's use is no longer tied to just one CLI command.